tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -182,11 +183,16 @@ public void splitRegionAndScatter(List<byte[]> splitKeys) {
     logger.info(String.format("split key's size is %d", splitKeys.size()));
     List<Key> rawKeys =
         splitKeys.parallelStream().map(RowKey::toRawKey).collect(Collectors.toList());
-    rawKeys.forEach(this::splitRegionAndScatter);
+    TLongArrayList regionIds = new TLongArrayList();
+    rawKeys.forEach(key -> splitRegionAndScatter(key, regionIds));
+    // TODO: adding a conf let user to decide wait or not.
```

Comment:
i recommend to add it now, it's simple.

Reply:
My plan is to do it with `enable region pre-split`. What do you think?